### PR TITLE
plan: fix index double read

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@hanfei1991 @zimulala PTAL

File: plan/new_plan_test.go [hunk 1]
```diff
@@ -903,6 +903,35 @@ func (s *testPlanSuite) TestConstantFolding(c *C) {
...
+	cases := []struct {
+		columnNames []string
+		indexNames  []string
+		indexLens   []int
+		isCovering  bool
+	}{
+		{[]string{"a"}, []string{"a"}, []int{-1}, true},
+		{[]string{"a"}, []string{"a", "b"}, []int{-1, -1}, true},
+		{[]string{"a", "b"}, []string{"b", "a"}, []int{-1, -1}, true},
+		{[]string{"a", "b"}, []string{"b", "c"}, []int{-1, -1}, false},
```
Comment (zimulala) [inline plan/new_plan_test.go]: Add the test case as following: `{[]string{"a", "b"}, []string{"a", "c"}, []int{-1, -1}, false},`

File: executor/new_executor_xapi.go [hunk 2]
```diff
@@ -269,7 +288,6 @@ func (e *NewXSelectIndexExec) doIndexRequest() (*xapi.SelectResult, error) {
 	concurrency := 1
 	if !e.indexPlan.DoubleRead {
 		concurrency = defaultConcurrency
-		selIdxReq.IndexInfo.Columns = xapi.ColumnsToProto(e.indexPlan.Columns, false)
```
Comment (shenli) [inline executor/new_executor_xapi.go]: Why remove this?

Comment (coocood) [inline executor/new_executor_xapi.go]: This was added mistakenly.

**Comment by shenli** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**